test/dependency-manager/validation.test.ts

```diff
@@ -423,22 +432,23 @@ services:
       } catch (e: any) {
         err = e;
       }
-      expect(err).instanceOf(ValidationErrors)
+      expect(err).instanceOf(ValidationErrors);
       const errors = JSON.parse(err.message) as ValidationError[];
       expect(errors).lengthOf(3);
       expect(errors.map(e => e.path)).members([
         'services.stateful-app.depends_on',
         'services.api.depends_on',
-        'services.backend.depends_on'
-      ])
+        'services.backend.depends_on',
+      ]);
+      expect(process.exitCode).eq(1);
```

Comment:
This is a great call to check exit code in the tests. However, I'm not sure this is the correct place for it. We want to test e2e against the base command finally/catch. 

Include a check if the `commands/register.test.ts` and `validate.test.ts` where we are actually running the oclif commands.